submitted_models/costar_husky_sensor_config_2/launch/spawner.rb

```diff
@@ -0,0 +1,85 @@
+def spawner(_name, _modelURI, _worldName, _x, _y, _z, _roll, _pitch, _yaw)
+  <<-HEREDOC
+  <plugin name="ignition::launch::GazeboFactory"
+          filename="libignition-launch-gazebo-factory.so">
+    <name>#{_name}</name>
+    <allow_renaming>false</allow_renaming>
+    <pose>#{_x} #{_y} #{_z + 0.2} #{_roll} #{_pitch} #{_yaw}</pose>
+    <world>#{_worldName}</world>
+    <is_performer>true</is_performer>
+    <sdf version='1.6'>
+    <include>
+      <name>#{_name}</name>
+      <uri>#{_modelURI}</uri>
+      <!-- Diff drive -->
+      <plugin filename="libignition-gazebo-diff-drive-system.so"
+              name="ignition::gazebo::systems::DiffDrive">
+        <left_joint>front_left_wheel_joint</left_joint>
+        <left_joint>rear_left_wheel_joint</left_joint>
+        <right_joint>front_right_wheel_joint</right_joint>
+        <right_joint>rear_right_wheel_joint</right_joint>
+        <wheel_separation>#{0.45649 * 1.5}</wheel_separation>
+        <wheel_radius>0.1651</wheel_radius>
+        <topic>/model/#{_name}/cmd_vel_relay</topic>
+      </plugin>
+      <!-- Publish robot state information -->
+      <plugin filename="libignition-gazebo-pose-publisher-system.so"
+        name="ignition::gazebo::systems::PosePublisher">
+        <publish_link_pose>true</publish_link_pose>
+        <publish_sensor_pose>true</publish_sensor_pose>
+        <publish_collision_pose>false</publish_collision_pose>
+        <publish_visual_pose>false</publish_visual_pose>
+        <publish_nested_model_pose>#{$enableGroundTruth}</publish_nested_model_pose>
+      </plugin>
+      <!-- Battery plugin -->
+      <plugin filename="libignition-gazebo-linearbatteryplugin-system.so"
+        name="ignition::gazebo::systems::LinearBatteryPlugin">
+        <battery_name>linear_battery</battery_name>
+        <voltage>12.694</voltage>
+        <open_circuit_voltage_constant_coef>12.694</open_circuit_voltage_constant_coef>
+        <open_circuit_voltage_linear_coef>-3.1424</open_circuit_voltage_linear_coef>
+        <initial_charge>78.4</initial_charge>
+        <capacity>78.4</capacity>
+        <resistance>0.061523</resistance>
+        <smooth_current_tau>1.9499</smooth_current_tau>
+        <power_load>6.6</power_load>
+        <start_on_motion>true</start_on_motion>
+      </plugin>
+      <!-- Gas Sensor plugin -->"
+      <plugin filename="libGasEmitterDetectorPlugin.so"
+        name="subt::GasDetector">
+        <topic>/model/#{_name}/gas_detected</topic>
+        <update_rate>10</update_rate>
+        <type>gas</type>
+      </plugin>
+      <plugin filename="libignition-gazebo-breadcrumbs-system.so"
+        name="ignition::gazebo::systems::Breadcrumbs">
+        <topic>/model/#{_name}/breadcrumb/deploy</topic>
+        <max_deployments>12</max_deployments>"
+        <breadcrumb>"
+          <sdf version="1.6">
+            <model name="#{_name}__breadcrumb__">
+              <pose>-1.2 0 0 0 0 0</pose>
```

Comment:
After testing these out for a little bit, a 1.2 m offset seems a little too far from the robot. If the robot's rear is right up against a wall, deploying the breadcrumb could put the breadcrumb inside the wall, or worse, outside the cave. It could also be a problem if there is another robot right behind a robot deploying a breadcrumb.

Reply:
Would it be possible to make the breadcrumb model not collide with the robot model, and spawn the breadcrumb inside the robot?

Reply:
The Collision bitmasks feature was added in ign-physics2, so we can try to backport it to ign-physics1. With that, we can make breadcrumbs not collide with any robot. I don't think we can easily make it so that it wouldn't collide only with the robot that spawned it. 

Would decreasing the offset not work?

Reply:
I was just thinking creatively. I'll decrease the offset.

Reply:
Adjusted spawn location: a1a886b42f913692dfb8fe9d099ef21ef4f22586